src/main/formats/Vab.cpp

```diff
@@ -127,43 +127,33 @@ bool Vab::GetInstrPointers() {
     uint32_t totalVAGSize = 0;
     VGMHeader *vagOffsetHdr = AddHeader(offVAGOffsets, 2 * 256, L"VAG Pointer Table");
 
-    uint32_t vagStartOffset = GetShort(offVAGOffsets) * 8;
-    vagOffsetHdr->AddSimpleItem(offVAGOffsets, 2, L"VAG Size /8 #0");
-    totalVAGSize = vagStartOffset;
+    uint32_t vagStartOffset = offVAGOffsets + 2 * 256;
+    uint32_t vagOffset = vagStartOffset;
 
     for (uint32_t i = 0; i < numVAGs; i++) {
-      uint32_t vagOffset;
-      uint32_t vagSize;
+      uint32_t vagSize = GetShort(offVAGOffsets + i * 2) * 8;
 
-      if (i == 0) {
-        vagOffset = vagStartOffset;
-        vagSize = GetShort(offVAGOffsets + (i + 1) * 2) * 8;
-      }
-      else {
-        vagOffset = vagStartOffset + vagLocations[i - 1].offset + vagLocations[i - 1].size;
-        vagSize = GetShort(offVAGOffsets + (i + 1) * 2) * 8;
-      }
-
-      swprintf(name, 256, L"VAG Size /8 #%u", i + 1);
-      vagOffsetHdr->AddSimpleItem(offVAGOffsets + (i + 1) * 2, 2, name);
+      swprintf(name, 256, L"VAG Size /8 #%u", i);
+      vagOffsetHdr->AddSimpleItem(offVAGOffsets + i * 2, 2, name);
 
       if (vagOffset + vagSize <= nEndOffset) {
-        vagLocations.push_back(SizeOffsetPair(vagOffset, vagSize));
+        vagLocations.emplace_back(SizeOffsetPair(vagOffset, vagSize));
```

Comment:
It's still a little different from my intention.

```suggestion
        vagLocations.emplace_back(vagOffset, vagSize);
```

 I should have used a suggestion style comment, but I was unfamiliar with reviews and overlooked them.

Reply:
This also draws my attention to the fact that this pair is badly named. It should be OffsetSizePair because of the ordering of inputs... But I'll follow up with renaming that rather than do it now.